### ISSUE TITLE
Ruby Paketo Buildpack should support Passenger apps

### ISSUE THREAD

**Issue by kvedurmu** [paketo-community/ruby#3]
**Background**
Passenger is a pretty popular Ruby web server, and it would be great if Paketo Buildpacks could support building apps that use Passenger. I'm throwing this feature request into the MRI CNB repo as we don't have a language family Ruby repo just yet. 

Notes:
- See https://github.com/paketo-community/mri/issues/18 for a proposal for how we can support generic web server start commands

Once this feature is complete, the Language Family Ruby buildpack ordering should look something like:
```
[[order]]
.........
  [[order.group]]
    id = "paketo-community/mri"
    version = "0.0.149"

  [[order.group]]
    id = "paketo-community/bundler"
    version = "0.0.137"

  [[order.group]]
    id = "paketo-community/bundle-install"
    version = "0.0.42"

  [[order.group]]
    id = "paketo-community/passenger"
    version = "VERSION"

  [[order.group]]
    id = "paketo-buildpacks/procfile"
    version = "1.3.10"
    optional = true
.................
```

**Comment by kvedurmu** [comment]
This looks great! I was able to build a rails app that uses passenger (https://github.com/kvedurmu/paketo-samples/tree/master/ruby/passenger/hello_rails) using both the standalone and NGINX Passenger engine. I did run into a separate issue around the Passenger Buildpack not supporting port config defined in the `Passengerfile.json`, but I think this first pass is great and we could tackle that in a separate issue. 

```
pack build rails-passenger-app --buildpack gcr.io/paketo-buildpacks/nodejs --buildpack gcr.io/paketo-community/ruby
full-cf: Pulling from paketo-buildpacks/builder
........
===> DETECTING
6 of 7 buildpacks participating
paketo-buildpacks/node-engine   0.0.260
paketo-buildpacks/npm           0.1.79
paketo-community/mri            0.0.160
paketo-community/bundler        0.0.150
paketo-community/bundle-install 0.0.57
paketo-community/passenger      0.0.1
===> ANALYZING
Restoring metadata for "paketo-buildpacks/node-engine:node" from app image
Restoring metadata for "paketo-community/mri:mri" from app image
Restoring metadata for "paketo-community/bundler:bundler" from app image
Restoring metadata for "paketo-community/bundle-install:gems" from app image
Restoring metadata for "paketo-community/passenger:curl" from app image
===> RESTORING
Restoring data for "paketo-buildpacks/node-engine:node" from cache
Restoring data for "paketo-community/mri:mri" from cache
Restoring data for "paketo-community/bundler:bundler" from cache
===> BUILDING
Node Engine Buildpack 0.0.260
  Resolving Node Engine version
    Candidate version sources (in priority order):
       -> "*"

    Selected Node Engine version (using ): 10.22.0

  Reusing cached layer /layers/paketo-buildpacks_node-engine/node

NPM Buildpack 0.1.79
  Resolving installation process
    Process inputs:
      node_modules      -> "Not found"
      npm-cache         -> "Not found"
      package-lock.json -> "Not found"

    Selected NPM build process: 'npm install'

  Executing build process
    Running 'npm install --unsafe-perm --cache /layers/paketo-buildpacks_npm/npm-cache'
      Completed in 27.094s

  Configuring environment
    NPM_CONFIG_LOGLEVEL   -> "error"
    NPM_CONFIG_PRODUCTION -> "true"
    PATH                  -> "$PATH:/layers/paketo-buildpacks_npm/modules/node_modules/.bin"
MRI Buildpack 0.0.160
  Resolving MRI version
    Candidate version sources (in priority order):
      <unknown> -> "2.6.5"
      <unknown> -> "*"

    Selected MRI version (using <unknown>): 2.6.5

  Reusing cached layer /layers/paketo-community_mri/mri

Bundler Buildpack 0.0.150
  Resolving Bundler version
    Candidate version sources (in priority order):
      <unknown> -> "*"
      <unknown> -> "*"

    Selected Bundler version (using <unknown>): 2.1.4

  Reusing cached layer /layers/paketo-community_bundler/bundler

Bundle Install Buildpack 0.0.57
  Executing build process
    Running 'bundle config path /layers/paketo-community_bundle-install/gems'
    Running 'bundle install'
      Completed in 3m25.408s

  Configuring environment
    BUNDLE_PATH -> "/layers/paketo-community_bundle-install/gems"

Passenger Buildpack 0.0.1
  Executing build process
    Installing cURL 7.71.1
      Completed in 416ms

  Assigning launch processes
    web: bundle exec passenger start --port ${PORT:-3000}

===> EXPORTING
Reusing layer 'paketo-buildpacks/node-engine:node'
Adding layer 'paketo-buildpacks/npm:modules'
Reusing layer 'paketo-community/mri:mri'
Reusing layer 'paketo-community/bundler:bundler'
Adding layer 'paketo-community/bundle-install:gems'
Reusing layer 'paketo-community/passenger:curl'
Adding 1/1 app layer(s)
Reusing layer 'launcher'
Adding layer 'config'
Adding label 'io.buildpacks.lifecycle.metadata'
Adding label 'io.buildpacks.build.metadata'
Adding label 'io.buildpacks.project.metadata'
*** Images (c93f818182b2):
      index.docker.io/library/rails-passenger-app:latest
Reusing cache layer 'paketo-buildpacks/node-engine:node'
Adding cache layer 'paketo-buildpacks/npm:npm-cache'
Reusing cache layer 'paketo-community/mri:mri'
Reusing cache layer 'paketo-community/bundler:bundler'
....
docker run -e PORT=3000 -e RAILS_ENV=development -p 3000:3000 rails-passenger-app
 --> Downloading a Phusion Passenger agent binary for your platform

 --> Installing Nginx 1.18.0 engine

--------------------------

 [passenger_native_support.so] trying to compile for the current user (vcap) and Ruby interpreter...
     (set PASSENGER_COMPILE_NATIVE_SUPPORT_BINARY=0 to disable)
     Compilation successful. The logs are here:
     /tmp/passenger_native_support-n02f4e.log
 [passenger_native_support.so] successfully loaded.
=============== Phusion Passenger Standalone web server started ===============
PID file: /workspace/tmp/pids/passenger.3000.pid
Log file: /workspace/log/passenger.3000.log
Environment: development
Accessible via: http://0.0.0.0:3000/
```
